x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -0,0 +1,323 @@
+import { once } from 'lodash';
+import type { ActionTypeExecutorResult } from '@kbn/actions-plugin/common';
+import type {
+  CrowdstrikeGetAgentsResponse,
+  CrowdstrikeGetAgentsParams,
+} from '@kbn/stack-connectors-plugin/common/crowdstrike/types';
+import {
+  CROWDSTRIKE_CONNECTOR_ID,
+  SUB_ACTION,
+} from '@kbn/stack-connectors-plugin/common/crowdstrike/constants';
+import type {
+  CommonResponseActionMethodOptions,
+  ProcessPendingActionsMethodOptions,
+} from '../../..';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import type { CrowdstrikeConnectorExecuteOptions } from './types';
+import { stringify } from '../../../../utils/stringify';
+import { ResponseActionsClientError } from '../errors';
+import type { ActionDetails, LogsEndpointAction } from '../../../../../../common/endpoint/types';
+import type { IsolationRouteRequestBody } from '../../../../../../common/api/endpoint';
+import type {
+  ResponseActionsClientOptions,
+  ResponseActionsClientWriteActionRequestToEndpointIndexOptions,
+  ResponseActionsClientValidateRequestResponse,
+} from '../lib/base_response_actions_client';
+import { ResponseActionsClientImpl } from '../lib/base_response_actions_client';
+
+export type CrowdstrikeActionsClientOptions = ResponseActionsClientOptions & {
+  connectorActions: ActionsClient;
+};
+
+export class CrowdstrikeActionsClient extends ResponseActionsClientImpl {
+  protected readonly agentType: ResponseActionAgentType = 'crowdstrike';
+  private readonly connectorActionsClient: ActionsClient;
+  private readonly getConnector: () => Promise<ConnectorWithExtraFindData>;
+
+  constructor({ connectorActions, ...options }: CrowdstrikeActionsClientOptions) {
+    super(options);
+    this.connectorActionsClient = connectorActions;
+
+    this.getConnector = once(async () => {
+      let connectorList: ConnectorWithExtraFindData[] = [];
+
+      try {
+        connectorList = await this.connectorActionsClient.getAll();
+      } catch (err) {
+        throw new ResponseActionsClientError(
+          `Unable to retrieve list of stack connectors: ${err.message}`,
+          // failure here is likely due to Authz, but because we don't have a good way to determine that,
+          // the `statusCode` below is set to `400` instead of `401`.
+          400,
+          err
+        );
+      }
+      const connector = connectorList.find(({ actionTypeId, isDeprecated, isMissingSecrets }) => {
+        return actionTypeId === CROWDSTRIKE_CONNECTOR_ID && !isDeprecated && !isMissingSecrets;
+      });
+
+      if (!connector) {
+        throw new ResponseActionsClientError(
+          `No Crowdstrike stack connector found`,
+          400,
+          connectorList
+        );
+      }
+
+      this.log.debug(`Using Crowdstrike stack connector: ${connector.name} (${connector.id})`);
+
+      return connector;
+    });
```

Comment:
FYI: this will need to get adjusted once [my PR ](https://github.com/elastic/kibana/pull/179388)merges